src/org/sosy_lab/java_smt/solvers/boolector/BoolectorAbstractProver.java

```diff
@@ -49,7 +49,7 @@ protected BoolectorAbstractProver(
     this.manager = manager;
     this.creator = creator;
     this.btor = btor;
-    terminationCallback = shutdownNotifier::shouldShutdown;
+    terminationCallback = proverShutdownManager.getNotifier()::shouldShutdown;
```

Comment:
Just for comparison to above: Here, we keep the same notifier and reuse it for all callback-calls.

Reply:
We build a new manager in every prover creation (see [here](https://github.com/sosy-lab/java-smt/pull/489/files#diff-3ef1d1629e51825e4d06bdc6fe00b365066088658fd72432476e4e5d1f50c220R67)) and then use this new managers notifier to request shutdowns. So we always use the one notifier bound to the one manager of the prover (that is a child of the contexts `ShutdownNotifier`)

Reply:
This is now outdated due to the changes.